src/router/advanced_router.py

```diff
@@ -47,6 +50,11 @@ def execute(
             translated_batch[lang] = combined_df.sort_index()
         
         return translated_batch
+    
+    def translate_batch(self, api: ITranslateAPI, batch: pd.DataFrame, column_names: List[str]) -> Dict[str, pd.DataFrame]:
+        if batch.empty:
+            return {lang: pd.DataFrame() for lang in api.to_languages}
```

Comment:
Nice. Much cleaner now, validation done before the api call instead of within the api.